File: bsp4rs/src/com/jetbrains/bsp/bsp4rs/bsp/task_finish_params.rs
use serde::{Deserialize, Serialize};

use crate::*;

#[derive(Debug, PartialEq, Eq, Clone, Default, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TaskFinishParams {
    /** Unique id of the task with optional reference to parent task id */
    #[serde(default)]
    pub task_id: TaskId,
    /** Timestamp of when the event started in milliseconds since Epoch. */
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub event_time: Option<i64>,
    /** Message describing the task. */
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub message: Option<String>,
    /** Task completion status. */
    #[serde(default)]
    pub status: StatusCode,
    /** Optional metadata about the task.
    Objects for specific tasks like compile, test, etc are specified in the protocol. */
    #[serde(default, flatten, skip_serializing_if = "Option::is_none")]
    pub data: Option<TaskFinishData>,
}

File: bsp4kt/src/org/jetbrains/bsp/bsp4kt/TaskFinishParams.kt
package org.jetbrains.bsp.bsp4kt

import kotlinx.serialization.Serializable
import kotlinx.serialization.SerialName
import kotlinx.serialization.json.JsonElement

@Serializable
data class TaskFinishParams(
  val taskId: TaskId,
  val originId: Identifier? = null,
  val eventTime: Long? = null,
  val message: String? = null,
  val status: StatusCode,
  val data: TaskFinishData? = null
)
